crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -56,10 +58,38 @@ pub async fn run() -> anyhow::Result<()> {
 }
 
 impl KvRedisProvider {
+    pub fn name() -> &'static str {
+        "keyvalue-redis-provider"
+    }
+
     pub async fn run() -> anyhow::Result<()> {
-        let HostData { config, .. } = load_host_data().context("failed to load host data")?;
+        let HostData {
+            config,
+            otel_config,
+            structured_logging,
```

Comment:
Do these changes need to be made on each provider that wants to initialize tracing/logging?

Reply:
Yup, they do -- See this bit in the PR/commit:

> set_global_default should not be called from libraries, so the configure_observability() function needed to be modified to ensure that configured subscribers were passed back out (as tracing::Dispatchs), and usable from the calling code in a binary context.

Everything seems to be pointing to the fact that we should have never been calling `set_global_default` from inside the SDK/lib side to begin with -- they were interfering with each other.

All the other providers need to do this as well

Reply:
As I [noted here](https://github.com/wasmCloud/wasmCloud/pull/2378#issuecomment-2186777538):

> NOTE: PR is failing tests due to https://github.com/wasmCloud/wasmCloud/pull/2379 -- once that merges and this PR is rebased we should be good to go -- please feel free to review the code that is here (not all providers are updated anyway, but the changes will look identical).

If the direction looks good, I can make this change to all the providers fairly quickly!

Reply:
Can we have a helper function in the sdk that handles init of logging?

Reply:
Ah so we can't have it be a function but we *could* have a macro (that must get called from binary code somewhere). The problem here is that we can't... call the function that actually completes the setup from the SDK (as it's gonna be in multiple places/that's the problem)!

Maybe something like `wasmcloud_provider_sdk::init_tracing!("name-of-your-provider")` ?

Reply:
basically, the `dispatch.try_init()` needs to *not* be called from library code. Everything else is fair game -- but the provider needs to hold on to the guard anyway as well for the duration of when it's running.

Reply:
Oh yeah I'd be cool with a macro 👍🏻 Just something that simplifies the couple of extra steps, as long as the functions you have to call are documented that'll work fine

Reply:
Done! Macro has been added, and all providers updated